### PR TITLE
Update Amplify Android version to 2.14.10

### PR DESCRIPTION
**Pull request by tylerjroach**
#### Description of changes:
Update referenced Amplify Android Version.

We also removed Kotlin SDK Developer Preview callout as we are now using GA version

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [x] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by mattcreaser — DISMISSED** [review]

**Review by chrisbonifacio — APPROVED** [review]

File: src/constants/versions.ts
Comment (hbuchel) [inline src/constants/versions.ts:1]: Resetting these formatting changes should let your build pass; not sure if intended or maybe you have some prettier settings auto updating it?.

**Review by hbuchel — COMMENTED** [review]
Left a comment about format changes to versions.ts

File: src/constants/versions.ts
Comment (tylerjroach) [reply src/constants/versions.ts:1]: Thanks, I didn't purposefully change these and didn't notice this. Fixing.

Comment (tylerjroach) [reply src/constants/versions.ts:1]: pushed update

**Review by hbuchel — APPROVED** [review]

**Review by mattcreaser — APPROVED** [review]

**Pull request merged**